### PR TITLE
wc_RsaKeyToPublicDer extension

### PR DESCRIPTION
**Pull request by ghs-sec**
Added function wc_RsaKeyToPublicDer_ex, enabling user to chose the DER header or not. This option was offered by static function SetRsaPublicKey but not accessible to the user

### CODE REVIEW

**Comment by SparkiDev** [comment]
ok to test

**Comment by JacobBarthelmeh** [comment]
Jenkins Findings:

- Macro guard for HAVE_SELFTEST (a CAVP build) needed around 

```
wolfcrypt/src/asn.c:11907:5: error: no previous prototype for ‘wc_RsaKeyToPublicDer_ex’ [-Werror=missing-prototypes]
```

- Similar FIPS build guard around the API's use for "Fips Check" and "FIPSv2 Regression Check" tests

```
wolfcrypt/src/asn.c:11907:5: error: no previous prototype for ‘wc_RsaKeyToPublicDer_ex’ [-Werror=missing-prototypes]
```

**Comment by ghs-sec** [comment]
Sorry, I'm not familiar with developing in wolfssl. I don't understand what the issue is. I assumed that prototyping and declaring the extended api where the "normal" api is prototyped/declared would not cause test failures if the "normal" one was passing the tests.

File: wolfssl/wolfcrypt/rsa.h [hunk 1]
```diff
@@ -353,6 +353,7 @@ WOLFSSL_API int wc_RsaExportKey(RsaKey* key,
                                 byte* q, word32* qSz);
 
 WOLFSSL_API int wc_RsaKeyToPublicDer(RsaKey*, byte* output, word32 inLen);
+WOLFSSL_API int wc_RsaKeyToPublicDer_ex(RsaKey* key, byte* output, word32 inLen, int with_header);
```
Comment (dgarske) [inline wolfssl/wolfcrypt/rsa.h:356]: For FIPS the rsa.h header is different (the certified version) so this API def won't exist. I recommend placing this new API def into asn_public.h instead. Or you will need to wrap the new asn.c in `#if !defined(HAVE_FIPS) && !defined(HAVE_SELFTEST)`.

**Comment by ghs-sec** [comment]
Ok I moved the prototype as you suggested

**Review by dgarske — APPROVED** [review]

**Comment by dgarske** [comment]
Hi @ghs-sec ,

In order to accept this PR we will need a signed contributor agreement. Please send an email to support@wolfssl.com and reference this ticket. 

Thanks,
David Garske, wolfSSL

**Comment by dgarske** [comment]
Hi @ghs-sec ,

I have gone ahead and put up a fix into this PR https://github.com/wolfSSL/wolfssl/pull/4261

Let me know if you have any issues. Closing this PR.

Thanks,
David Garske, wolfSSL

**Pull request closed without merging**